### PR TITLE
refactor: split pane

### PR DESCRIPTION
**Pull request by wadjih-bencheikh18**
closes https://github.com/zakodium-oss/analysis-ui-components/issues/209

### CODE REVIEW

File: src/components/split-pane/SplitPane.tsx [hunk 1]
```diff
@@ -99,7 +99,6 @@ export function SplitPane(props: SplitPaneProps) {
 
   const splitterRef = useRef<HTMLDivElement>(null);
   const { onMouseDown } = useSplitPaneSize({
-    mainSide,
```
Comment (targos) [inline src/components/split-pane/SplitPane.tsx:102]: I think the hook should still depend on this somehow. It doesn't work now when the value is `'end'`:
https://deploy-preview-228--analysis-ui-components.netlify.app/stories/?arg-r-mainSide=end&story=components--splitpane--control
https://deploy-preview-228--analysis-ui-components.netlify.app/

File: package.json [hunk 2]
```diff
@@ -60,14 +59,15 @@
...
-    "spc-parser": "^0.5.2",
+    "spc-parser": "^0.6.0",
     "tinycolor2": "^1.4.2",
     "use-resize-observer": "^9.0.2",
     "wdf-parser": "^0.2.1"
   },
   "devDependencies": {
     "@babel/core": "^7.19.3",
     "@babel/eslint-parser": "^7.19.1",
+    "@emotion/react": "^11.10.4",
```
Comment (targos) [inline package.json:70]: This has to stay in `dependencies`

File: package.json [hunk 3]
```diff
@@ -60,14 +59,15 @@
     "react-dropzone": "^14.2.2",
     "react-error-boundary": "^3.1.4",
     "react-shadow": "^19.0.3",
-    "spc-parser": "^0.5.2",
+    "spc-parser": "^0.6.0",
```
Comment (targos) [inline package.json:62]: Please keep version 0.5.2. The new version has breaking changes.

File: src/components/split-pane/SplitPane.tsx (hunk 1 above)
Comment (wadjih-bencheikh18) [reply src/components/split-pane/SplitPane.tsx]: fixed thank you

File: package.json (hunk 2 above)
Comment (wadjih-bencheikh18) [reply package.json]: done sorry about that i didn't pay attention

File: package.json (hunk 3 above)
Comment (wadjih-bencheikh18) [reply package.json]: okey

**Comment by targos** [comment]
Nice work 👍🏻

**Pull request merged**